lib/load-simple-icons.js

```diff
@@ -5,18 +5,27 @@ const { svg2base64 } = require('./svg-helpers')
 
 function loadSimpleIcons() {
   const simpleIcons = {}
+  // As of v5 the exported keys are the svg slugs
+  // Historically, Shields has supported logo specification via
+  // name, name with spaces replaced by hyphens, and partially slugs
+  // albeit only in cases where the slug happened to match one of those.
+  // For backwards compatibility purposes we now support all three, but
+  // do not broadcast the support for by-title references due to our strong
+  // preference to steer users towards using the actual slugs.
+  // https://github.com/badges/shields/pull/6591
+  // https://github.com/badges/shields/issues/4273
   Object.keys(originalSimpleIcons).forEach(key => {
-    const k = key.toLowerCase().replace(/ /g, '-')
-    simpleIcons[k] = originalSimpleIcons[key]
-    simpleIcons[k].base64 = {
-      default: svg2base64(
-        simpleIcons[k].svg.replace('<svg', `<svg fill="#${simpleIcons[k].hex}"`)
-      ),
-      light: svg2base64(
-        simpleIcons[k].svg.replace('<svg', `<svg fill="whitesmoke"`)
-      ),
-      dark: svg2base64(simpleIcons[k].svg.replace('<svg', `<svg fill="#333"`)),
+    const icon = originalSimpleIcons[key]
+    const title = icon.title.toLowerCase()
+    const legacyTitle = title.replace(/ /g, '-')
+    icon.base64 = {
+      default: svg2base64(icon.svg.replace('<svg', `<svg fill="#${icon.hex}"`)),
+      light: svg2base64(icon.svg.replace('<svg', `<svg fill="whitesmoke"`)),
+      dark: svg2base64(icon.svg.replace('<svg', `<svg fill="#333"`)),
     }
+    simpleIcons[key] = icon
+    simpleIcons[title] = icon
+    simpleIcons[legacyTitle] = icon
```

Comment:
Shall we change the order here.
i.e: if we're going to say the SI slug is the canonical ID and the others are fallbacks, then in the case of a collision like "hive" then if we set the SI slug last then that value is guaranteed to "win" the clash here.

Reply:
Good shout. Actually needed to incorporate the key check after all as originally proposed (just skipping instead of throwing exception). The order of the assignment within a given iteration doesn't matter so much, but more about needing the conditional checks otherwise the value would be determined by the last iteration that contains a colliding key. 32554a2a5b7471630d2dff5f61dd9e443b437fef